url0980c8326f_TobiasScharnowski_Chess_tgz-pJ8-chess_StartJ8/src/chess/piece/impl/King.java

```diff
@@ -0,0 +1,158 @@
+package chess.piece.impl;
+
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+import chess.ChessCoord;
+import chess.ChessRules;
+import chess.PieceType;
+import chess.Position;
+import chess.impl.Coords;
+import chess.move.Move;
+import chess.move.impl.CastlingMove;
+import chess.piece.ChessColor;
+import chess.piece.Piece;
+
+/**
+ * Class representing the default Chess King piece
+ * 
+ * @author Tobias Scharnowski
+ * 
+ */
+public class King extends AbstractPiece {
+
+    public King(Position pos, ChessColor cc, ChessCoord initialCoord) {
+        super(pos, PieceType.KING, cc, initialCoord);
+    }
+
+    @Override
+    protected List<Move> _generateMoves() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 17. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 78-95 into a separate method.